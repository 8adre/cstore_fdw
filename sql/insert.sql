--
-- Testing insert on cstore_fdw tables.
--

-- Settings to make the result deterministic
SET datestyle = "ISO, YMD";

CREATE FOREIGN TABLE cstore_table (a int) SERVER cstore_server;


-- test single row inserts fail
select count(*) from cstore_table;
insert into cstore_table values(1);
select count(*) from cstore_table;


-- test inserting from another table succeed
CREATE TABLE regular_table (a int);
select count(*) from regular_table;
insert into regular_table values(1);
select count(*) from regular_table;

insert into cstore_table select * from regular_table;
select count(*) from cstore_table;


drop table regular_table;
drop foreign table cstore_table;
