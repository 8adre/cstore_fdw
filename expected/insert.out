--
-- Testing insert on cstore_fdw tables.
--
-- Settings to make the result deterministic
SET datestyle = "ISO, YMD";
CREATE FOREIGN TABLE cstore_table (a int) SERVER cstore_server;
-- test single row inserts fail
select count(*) from cstore_table;
 count 
-------
     0
(1 row)

insert into cstore_table values(1);
ERROR:  operation is not supported
select count(*) from cstore_table;
 count 
-------
     0
(1 row)

-- test inserting from another table succeed
CREATE TABLE regular_table (a int);
select count(*) from regular_table;
 count 
-------
     0
(1 row)

insert into regular_table values(1);
select count(*) from regular_table;
 count 
-------
     1
(1 row)

insert into cstore_table select * from regular_table;
select count(*) from cstore_table;
 count 
-------
     1
(1 row)

drop table regular_table;
drop foreign table cstore_table;
